src/frontend/src/icp-eth/components/tokens/ManageTokensModal.svelte

```diff
@@ -90,6 +100,69 @@
 		}
 	};
 
+	let erc20Metadata: Erc20Metadata | undefined;
+
+	const saveErc20Token = async () => {
```

Comment:
If we are already there, I would rather like to have this in a service or at least everything we can extract. Or are you planing to refactor that afterwards?

Reply:
i was planning to do it afterwrds actually, so that we can move step by step, it seems the best way to have smaller PRs

what do you think? I refactor first?